tests/communication_tests.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <unistd.h>
+#include <tuple>
+#include "doctest.h"
+#include "pseudoterminal.h"
+#include "LinSerial.h"
+
+TEST_SUITE ("communication_tests")
+{
+    TEST_CASE ("read_single_line")
+    {
+        auto tunnel = pseudoterminal::setup();
+        LinSer::Serial writer(std::get<1>(tunnel).c_str());
```

Comment:
why use a tuple and not an array?
even better would be something like a class / struct with "endpoint1" and "endpoint2" so that code is less ambigious

Reply:
I refactored it into a class now.